raiden/transfer/architecture.py

```diff
@@ -402,3 +402,26 @@ def token_network_address(self) -> TokenNetworkAddress:
     @property
     def channel_identifier(self) -> ChannelID:
         return self.canonical_identifier.channel_identifier
+
+
+class Check:
+    """ A Check instance should be used when you want to test a boolean
+    and _also_ collect feedback when the test fails """
```

Comment:
Types would need to be added in this class. I would still call this `SuccessOrError` or something like that. `Check` is a very generic name